node/pkg/guardiansigner/amazonkms.go

```diff
@@ -0,0 +1,255 @@
+		return nil, fmt.Errorf("Failed to decode signature: %w", err)
+	}
+
+	// if s is greater than secp256k1HalfN, we need to substract secp256k1N from it
+	sBigInt := new(big.Int).SetBytes(s)
+	if sBigInt.Cmp(secp256k1HalfN) > 0 {
+		s = new(big.Int).Sub(secp256k1N, sBigInt).Bytes()
+	}
+
+	// r and s need to be 32 bytes in size
+	r = adjustBufferSize(r)
+	s = adjustBufferSize(s)
+
+	// AWS KMS does not provide the recovery id. But that doesn't matter too much, since we can
+	// attempt recovery id's 0 and 1, and in the process ensure that the signature is valid.
+	expectedPublicKey := a.PublicKey(ctx)
+	signature = append(r, s...)
+
+	// try recovery id 0
+	ecSigWithRecid := append(signature, []byte{0}...)
+	pubkey, _ := ethcrypto.SigToPub(hash[:], ecSigWithRecid)
+
+	if bytes.Equal(ethcrypto.CompressPubkey(pubkey), ethcrypto.CompressPubkey(&expectedPublicKey)) {
+		return ecSigWithRecid, nil
+	}
+
+	ecSigWithRecid = append(signature, []byte{1}...)
+	pubkey, _ = ethcrypto.SigToPub(hash[:], ecSigWithRecid)
+
+	// try recovery id 1
+	if bytes.Equal(ethcrypto.CompressPubkey(pubkey), ethcrypto.CompressPubkey(&expectedPublicKey)) {
+		return ecSigWithRecid, nil
+	}
+
+	// Reaching this return implies that it wasn't possible to generate a valid signature. This shouldn't
+	// happen, unless there is something seriously wrong with the KMS service.
+	return nil, fmt.Errorf("Failed to generate valid signature")
+}
+
+func (a *AmazonKms) PublicKey(ctx context.Context) ecdsa.PublicKey {
+	return a.publicKey
+}
+
+func (a *AmazonKms) Verify(ctx context.Context, sig []byte, hash []byte) (bool, error) {
+	timeoutCtx, cancel := context.WithTimeout(ctx, time.Second*15)
+	defer cancel()
+
+	// Use ethcrypto to recover the public key
+	recoveredPubKey, err := ethcrypto.SigToPub(hash, sig)
+
+	if err != nil {
+		return false, err
+	}
+
+	// Load the KMS signer's public key
+	kmsPublicKey := a.PublicKey(timeoutCtx)
+
+	return recoveredPubKey.Equal(kmsPublicKey), nil
+}
+
+// Return the signer type as "amazonkms".
+func (a *AmazonKms) TypeAsString() string {
+	return "amazonkms"
+}
+
+// https://bitcoin.stackexchange.com/questions/92680/what-are-the-der-signature-and-sec-format
+//  1. 0x30 byte: header byte to indicate compound structure
+//  2. one byte to encode the length of the following data
+//  3. 0x02: header byte indicating an integer
+//  4. one byte to encode the length of the following r value
+//  5. the r value as a big-endian integer
+//  6. 0x02: header byte indicating an integer
+//  7. one byte to encode the length of the following s value
+//  8. the s value as a big-endian integer
+func derSignatureToRS(signature []byte) ([]byte, []byte, error) {
+	var sigAsn1 asn1EcSig
+	_, err := asn1.Unmarshal(signature, &sigAsn1)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return sigAsn1.R.Bytes, sigAsn1.S.Bytes, nil
+}
+
+// adjustBufferSize takes an input buffer and
+// a) trims it down to 32 bytes, if the input length is greater than 32, or
```

Comment:
When I read "trim it down to 32 bytes" I would guess the least significant bytes are trimmed, but this trims from the front of the array. Maybe worth making this clearer in the comment?